examples/mamba/convert_hf_to_nanotron.py

```diff
@@ -0,0 +1,342 @@
+        raise ValueError("Model should be one of 130M, 370M, 790M, 1.4B, 2.8B")
+
+    save_path = Path(args.save_path)
+
+    parallel_config = ParallelismArgs(
+        dp=args.dp,
+        pp=args.pp,
+        tp=args.tp,
+        pp_engine=AllForwardAllBackwardPipelineEngine(),
+        tp_mode=TensorParallelLinearMode.ALL_REDUCE,
+        tp_linear_async_communication=False,
+    )
+
+    parallel_context = ParallelContext(
+        data_parallel_size=parallel_config.dp,
+        pipeline_parallel_size=parallel_config.pp,
+        tensor_parallel_size=parallel_config.tp,
+    )
+
+    # Set log log levels
+    logging_config = LoggingArgs(
+        log_level="info",
+        log_level_replica="info",
+    )
+
+    # Set log levels
+    set_ranks_logging_level(parallel_context=parallel_context, logging_config=logging_config)
+
+    d_model = None
+    num_hidden_layers = None
+    pretrained_model_name = None
+
+    if args.model == "130M":
+        d_model = 768
+        num_hidden_layers = 24
+        pretrained_model_name = "state-spaces/mamba-130m"
+    elif args.model == "370M":
+        d_model = 1024
+        num_hidden_layers = 48
+        pretrained_model_name = "state-spaces/mamba-370m"
+    elif args.model == "790M":
+        d_model = 1536
+        num_hidden_layers = 24
+        pretrained_model_name = "state-spaces/mamba-790m"
+    elif args.model == "1.4B":
+        d_model = 2048
+        num_hidden_layers = 48
+        pretrained_model_name = "state-spaces/mamba-1.4b"
+    elif args.model == "2.8B":
+        d_model = 2560
+        num_hidden_layers = 64
+        pretrained_model_name = "state-spaces/mamba-2.8b"
+
+    yaml_content = f"""
+    is_mamba_config: true
+    d_model: {d_model}
+    dtype: float32
+    fused_add_norm: true
+    is_mamba_config: true
+    num_hidden_layers: {num_hidden_layers}
+    pad_token_id: null
+    pad_vocab_size_multiple: 8
+    residual_in_fp32: true
+    rms_norm: true
+    rms_norm_eps: 1.0e-05
+    ssm_cfg: null
+    vocab_size: 50277
+    """
+
+    str_to_dtype = {
+        "float32": torch.float32,
+        "float64": torch.float64,
+        "complex64": torch.complex64,
+        "complex128": torch.complex128,
+        "float16": torch.float16,
+        "bfloat16": torch.bfloat16,
+        "uint8": torch.uint8,
+        "int8": torch.int8,
+        "int16": torch.int16,
+        "int32": torch.int32,
+        "int64": torch.int64,
+        "bool": torch.bool,
+    }
+    device = torch.device("cuda")
+
+    attrs = yaml.safe_load(yaml_content)
+    model_config = MambaModelConfig(**attrs)
+    
+    assert model_config.dtype == "float32", "Convert weights only in float32"
+
+    # Initiliaze Brrr model
+    model_config.vocab_size = _vocab_size_with_padding(
+            model_config.vocab_size,
+            pg_size=parallel_context.tp_pg.size(),
+            make_vocab_size_divisible_by=5, # So that every value of TP from 1 to 8 yield a vocab_size of 50280
```

Comment:
not sure I understand this comment. Why does TP care abt vocab size?

Reply:
`_vocab_size_with_padding`does this `multiple = make_vocab_size_divisible_by * pg_size` and then `after = int(ceil(orig_vocab_size / multiple) * multiple)`.  Not sure why this function is this way but I found that 5 gives 50280 for every value of TP. Maybe it is better that I hardcode the `model_config.vocab_size=50280` directly

Reply:
I think we should keep the value provided by vocab_size nad have vocab_size=50280 in the config? Is there any reason why we want to force it to be a multiple of 5 though?